ingress/controllers/nginx/controller.go

```diff
@@ -373,7 +524,12 @@ func (lbc *loadBalancerController) getServices(data map[string]string, proto api
 		var endps []nginx.UpstreamServer
 		targetPort, err := strconv.Atoi(svcPort[1])
 		if err != nil {
-			endps = lbc.getEndpoints(svc, intstr.FromString(svcPort[1]), proto)
+			for _, sp := range svc.Spec.Ports {
```

Comment:
nit: can you cache svcPort[0,1] in appropriately named local vars instead of re-indexing?